code/obj/item/cameras.dm

```diff
@@ -169,6 +170,298 @@ TYPEINFO(/obj/item/camera/large)
+		if (!victim_chosen)
+			victim_chosen = findSentientOnTurf(the_turf, /mob/living/silicon/robot)
+			victim_type = VICTIM_TYPE_CYBORG
+		// No borgs? Find an AI....
+		if (!victim_chosen)
+			victim_chosen = findSentientOnTurf(the_turf, /mob/living/silicon/ai)
+			victim_type = VICTIM_TYPE_AI
+		// No AI? Find a critter...
+		if (!victim_chosen)
+			victim_chosen = findSentientOnTurf(the_turf, /mob/living/critter)
+			victim_type = VICTIM_TYPE_CREATURE
+		// Oh yeah, object critters are still a thing.
+		if (!victim_chosen)
+			victim_chosen = findSentientOnTurf(the_turf, /obj/critter)
+			victim_type = VICTIM_TYPE_CREATURE
+		// why are bots considered obj/machinery...
+		if (!victim_chosen)
+			victim_chosen = findSentientOnTurf(the_turf, /obj/machinery/bot)
+			victim_type = VICTIM_TYPE_CREATURE
+		// Lost at 20 questions again! Get area card as fallback
+		if (!victim_chosen)
+			victim_type = VICTIM_TYPE_NOTHING
+
+		switch(victim_type)
+			if(VICTIM_TYPE_HUMAN)
+				// Let's dig into your mind and extract some things...
+				// You an antag we know? Just like areas they dont (or shouldnt) map 1:1 to griff. So sad! Atleast we dont need special_role back compat
+				var/mob/living/carbon/human/victim_human = victim_chosen
+				var/list/stg_antag_table = list()
+				stg_antag_table += list(ROLE_TRAITOR, /datum/playing_card/griffening/creature/mob/traitor)
+				stg_antag_table += list(ROLE_CHANGELING, /datum/playing_card/griffening/creature/mob/changeling)
+				stg_antag_table += list(ROLE_VAMPIRE, /datum/playing_card/griffening/creature/mob/vampire)
+				stg_antag_table += list(ROLE_WRAITH, /datum/playing_card/griffening/creature/mob/wraith) // Impressive!
+				stg_antag_table += list(ROLE_NUKEOP, /datum/playing_card/griffening/creature/mob/nukeop)
+				stg_antag_table += list(ROLE_WIZARD, /datum/playing_card/griffening/creature/mob/wizard) // lol
+				var/datum/mind/victim_mind = null
+				if (ismind(victim_human.mind))
+					victim_mind = victim_human.mind
+					for (var/list/compare_antag in stg_antag_table)
+						if(victim_mind && victim_mind.get_antagonist(compare_antag[0]))
+							card_data = new compare_antag[1]
+							break
+					//message_admins(ismind(victim_mind))
+					//message_admins(victim_mind?.assigned_role)
+					// No antag? What's your job? Sadly text2path ended up being too janky, so this is yet another table mess.
+					// Thankfully this time we can do it with a switch without too much redundancy to be painful
+					if(victim_mind)
+						switch(victim_mind.assigned_role)
+							if("Captain")
+								card_data = new /datum/playing_card/griffening/creature/mob/captain
+							if("Head of Personnel")
+								card_data = new /datum/playing_card/griffening/creature/mob/head_of_personnel
+							if("Head of Security")
+								card_data = new /datum/playing_card/griffening/creature/mob/head_of_security
+							if("Research Director")
+								card_data = new /datum/playing_card/griffening/creature/mob/head_of_security
```

Comment:
you forgot to change this to research director.

Reply:
Whoops. Let me quickly suggest-commit that.
```suggestion
								card_data = new /datum/playing_card/griffening/creature/mob/research_director
```